pkg/thing/interactors/publish_data.go

```diff
@@ -28,6 +29,31 @@ func (i *ThingInteractor) PublishData(authorization, thingID string, data []enti
 		return fmt.Errorf("error publishing data in broadcast mode: %w", err)
 	}
 
-	i.logger.Info("publish data message successfully sent")
+	err = i.publishSessionData(thingID, authorization, data)
+	if err != nil {
+		return fmt.Errorf("error publishing data to user sessions: %w", err)
+	}
+
+	return nil
+}
+
+func (i *ThingInteractor) publishSessionData(thingID, authorization string, data []entities.Data) error {
+	email, err := jwt.GetEmail(authorization)
+	if err != nil {
+		return err
+	}
+
+	sessionId, err := i.sessionStore.Get(email)
+	if err != nil {
+		return fmt.Errorf("error getting user session: %w", err)
+	}
+
+	if sessionId != "" {
```

Comment:
We should verify this instead of relying on the reliability of other services (this dependency is not mapped anywhere). But I agree with @esdrasjnr that we should return an error with context :point_down: 

```go
if sessionId == "" {
    return fmt.Errorf("error getting user sessionId: %w", errSessionIdIsEmpty)
    // return errSessionIdIsEmpty
}

// rest of the code

```

This would also require adding some test cases!

Reply:
It's not mandatory having a session for each user. The idea is that users can explicitly create a session when subscribing to their owned events.

e.g. if data is published on behalf of user `esdras@gmail.com` and he doesn't create a session for receiving them, the events will be published in broadcast-only mode. So, there's no reason to return an error.

Reply:
@esdrasjnr @lcbm 🆙